File: ListViews/ListDemo/app/src/main/java/edu/cs4730/ListDemo/InterActive_ListFragment.java
package edu.cs4730.ListDemo;

import java.util.ArrayList;
import java.util.List;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;

import androidx.annotation.NonNull;
import androidx.fragment.app.ListFragment;


import edu.cs4730.ListDemo.databinding.ListfragmentLayoutBinding;

public class InterActive_ListFragment extends ListFragment {
    String TAG = "InterActive_ListFragment";
	ListfragmentLayoutBinding binding;

	@Override
	public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {

		binding = ListfragmentLayoutBinding.inflate(inflater, container, false);
		ArrayAdapter<InterActive_DataModel> adapter = new InterActive_myArrayAdapter(requireActivity(),	getModel());
		setListAdapter(adapter);
		return binding.getRoot();
	}

	private List<InterActive_DataModel> getModel() {
		List<InterActive_DataModel> list = new ArrayList<InterActive_DataModel>();
		list.add(new InterActive_DataModel("Linux"));
		list.add(new InterActive_DataModel("Windows7"));
		list.add(new InterActive_DataModel("Suse"));
		list.add(new InterActive_DataModel("Eclipse"));
		list.add(new InterActive_DataModel("Ubuntu"));
		list.add(new InterActive_DataModel("Solaris"));
		list.add(new InterActive_DataModel("Android"));
		list.add(new InterActive_DataModel("iPhone"));
		// Initially select one of the items
		list.get(1).setSelected(true);
		return list;
	}

}

File: ListViews/ListDemo_kt/app/src/main/java/edu/cs4730/listdemo_kt/InterActive_ListFragment.kt
package edu.cs4730.listdemo_kt

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import androidx.fragment.app.ListFragment
import edu.cs4730.listdemo_kt.databinding.ListfragmentLayoutBinding

class InterActive_ListFragment : ListFragment() {
    var TAG = "InterActive_ListFragment"
    lateinit var binding: ListfragmentLayoutBinding

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?
    ): View {
        binding = ListfragmentLayoutBinding.inflate(inflater, container, false)
        val adapter: InterActive_myArrayAdapter = InterActive_myArrayAdapter(
            requireActivity(), model
        )
        listAdapter = adapter
        return binding.root
    }

    // Initially select one of the items
    private val model: List<InterActive_DataModel>
        get() {
            val list: MutableList<InterActive_DataModel> = ArrayList()
            list.add(InterActive_DataModel("Linux"))
            list.add(InterActive_DataModel("Windows7"))
            list.add(InterActive_DataModel("Suse"))
            list.add(InterActive_DataModel("Eclipse"))
            list.add(InterActive_DataModel("Ubuntu"))
            list.add(InterActive_DataModel("Solaris"))
            list.add(InterActive_DataModel("Android"))
            list.add(InterActive_DataModel("iPhone"))
            // Initially select one of the items
            list[1].isSelected = true
            return list
        }
}